core/templates/pages/blog-admin-page/blog-admin-page.component.spec.ts

```diff
@@ -0,0 +1,411 @@
+        }
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(BlogAdminPageComponent);
+    component = fixture.componentInstance;
+  });
+
+  beforeEach(() => {
+    blogAdminBackendApiService = TestBed.inject(BlogAdminBackendApiService);
+    adminTaskManagerService = TestBed.inject(AdminTaskManagerService);
+
+    startTaskSpy = spyOn(adminTaskManagerService, 'startTask')
+      .and.returnValue(null);
+    finishTaskSpy = spyOn(adminTaskManagerService, 'finishTask')
+      .and.returnValue(null);
+    spyOn(blogAdminBackendApiService, 'getDataAsync')
+      .and.resolveTo(blogAdminPageData);
+    confirmSpy = spyOn(mockWindowRef.nativeWindow, 'confirm');
+  });
+
+  it('should refresh roles form data when initialized', fakeAsync(() => {
+    expect(component.formData).toBe(undefined);
+
+    component.ngOnInit();
+    tick();
+
+    expect(component.formData.updateRole.newRole).toBe(null);
+    expect(component.formData.updateRole.username).toBe('');
+    expect(component.formData.removeEditorRole.username).toBe('');
+  }));
+
+  it('should set correct values for properties when initialized',
+    fakeAsync(() => {
+      expect(component.UPDATABLE_ROLES).toEqual({});
+      expect(component.roleToActions).toBe(undefined);
+
+      component.ngOnInit();
+      tick();
+
+      expect(component.UPDATABLE_ROLES).toEqual(
+        blogAdminPageData.updatableRoles);
+      expect(component.roleToActions).toEqual(
+        blogAdminPageData.roleToActions);
+    }));
+
+  it('should reload config properties when initialized', fakeAsync(() => {
+    expect(component.configProperties).toEqual({});
+
+    component.ngOnInit();
+    tick();
+
+    expect(component.configProperties).toEqual(
+      blogAdminPageData.configProperties);
+  }));
+
+  it('should return schema callback when calling ' +
+  '\'getSchemaCallback\'', () => {
+    let result = component.getSchemaCallback({type: 'bool'});
+    expect(result()).toEqual({type: 'bool'});
+  });
+
+  describe('when updating role of user ', () => {
+    it('should submit update role form successfully', fakeAsync(() => {
+      component.ngOnInit();
+      tick();
+      component.formData.updateRole.newRole = 'BLOG_ADMIN';
+      component.formData.updateRole.username = 'username';
+      spyOn(blogAdminBackendApiService, 'updateUserRoleAsync')
+        .and.returnValue(Promise.resolve());
+      spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(false);
+      component.submitUpdateRoleForm(component.formData.updateRole);
+
+      expect(startTaskSpy).toHaveBeenCalled();
+      expect(component.statusMessage).toBe('Updating User Role');
+      flushMicrotasks();
+      expect(component.statusMessage).toBe(
+        'Role of username successfully updated to BLOG_ADMIN');
+      expect(finishTaskSpy).toHaveBeenCalled();
+    }));
+
+    it('should not submit update role form if already a task is in queue',
```

Comment:
I'm not able to distinguish between the setup, pre-checks, action, and post-checks steps for this test. Please add a space after each step.

Reply:
Done.